content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,210 @@
+A chit value reflect the result from a single query of its associated transaction and becomes immutable afterwards, while confidence `d` can increase as the DAG grows by collecting more chits in its progeny.
+Additionally each node maintains its own local list of known nodes: Nu — nodes known to `u` is a subset of the set of all existing nodes: `Nu ⊆ N`.
+
+Each node implements an event-driven state machine, centered around a query that serves both to solicit votes on each transaction and to notify other nodes of the existence of newly discovered transactions.
+When a node `u` discovers a trancation `T` through a query, it starts a one-time query process by sampling k random peers and sending a message to them, after T is delivered to the node by onReceiveTx.
+Node `u` answers a query by checking whether each T' such that `T' *<— T` is currently prefered among competing transactions `∀T'' ∈ PT'` from conflict set P of T'.
+If every single ancestor T' fulfills this criterion, the transation is said to br strongly preferred and receives a yes-vote (1).
+A failure on this criterion at any T' yields a no-vote (0).
+When `u` accumulates k responses, it checks whether there are α yes-votes for T, and if so grants the chit value (cT = 1).
+This process will yield a labeling of the DAG with a chit value and associated confidence for each transaction T.
+
+Example of (chit,confidence) values:
+
+```text
+
+T1(1,6)
+├── T2(1,5)
+│    ├── T4(1,2) ── T8(1,1)
+│    │               │
+│    └────────────── T5(1,3)
+│                    └──── T9(1,1)
+│
+└── T3(0,0)
+    ├── T6(0,0)
+    └── T7(0,0)
+
+```
+
+Conflict set P for T1 has one element T1.
+Conflict set P for T2 and T3 has two elements {T2, T3}
+Conflict set P for T5, T6 and T9 has 3 elements {T6, T7, T9}
+
+Similar to Snowball, sampling in Avalanche will create a positive feedback for the preference of a single transaction in its conflict set.
+When a transaction has a larger confidence its decendants are likely to collect chits in the future compared to others with lower confidence.
+For example T2 has larger confidence than T3 and its decendants are more likely to collect chits in the future.
+
+Similar to Bitcoin, Avalanche leaves determining the acceptance point of a transaction to the application.
+Committing a transaction can be performed through a safe early commitment.
+For virtuous transactions, T is accepted when it is the only transaction in its confliict set and has a confidence no less than a threshold of β1.
+If a virtuous transaction fails to get accepted due to a problem with parents, it could be accepted if reissued with different parents.
+
+### Transaction Entangling
+
+How Avalance entangles transactions:
+
+```text
+init:
+    `Set T` = Ø set of known transactions (empty set)
+    Q = Ø set of queried transactions (empty set)
+
+generateTx:
+    edges = {T' <— T: T' ∈ parentSelection(`Set T`)}
+    T = Tx(data, edges)
+    onReceiveEvent(T)
+
+onReceiveEvent(t)
```

Comment:
```suggestion
onReceiveEvent(t):
```

Reply:
This is corrected https://github.com/optakt/optakt.github.io/pull/13/commits/e0e27e6be8c0aef4153af5e2add9722380c23a59#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98R116
Thanks!